Shogun derivation installs wrong CMake files (i.e. which point to inexistent /nix/store/path)

### Describe the bug
Using Shogun as a `builtInput` for another CMake-based project fails because the dependent project cannot properly find the shogun library (via its installed cmake files).

### Steps To Reproduce
Steps to reproduce the behavior:
1. Try to build a derivation based on shogun
2. Cmake files are wrong

### Expected behavior
Everything builds correctly

### Additional context
I am trying to build the following derivation
```nix
{ lib, stdenv, fetchFromGitHub, cmake, eigen, shogun }:
stdenv.mkDerivation rec {
  pname = "feat";
  version = "0.4.2";

  src = fetchFromGitHub {
    owner = "lacava";
    repo = "feat";
    rev = "${version}";
    sha256 = "sha256-mbvASAB3UQ3ObrVOmkLGVOdFtuAOcqotlQGxi2zSPXg=";
  };

  nativeBuildInputs = [ cmake ];

  buildInputs = [ eigen (shogun.overrideAttrs(old: rec {
    doCheck = false;
    cmakeFlags = old.cmakeFlags ++ [ "-DENABLE_TESTING=OFF" ];
  })) ];

  meta = with lib; {
    description = "FEAT is a feature engineering automation tool that learns new representations of raw data to improve classifier and regressor performance.";
    homepage = "https://lacava.github.io/feat";
    license = licenses.gpl3;
    platforms = platforms.all;
  };
}
```

However I get an error at the build generation step:
```
CMake Error at /nix/store/a6y70prh981r98hs1plk24r4rqzx335s-shogun-6.1.4-dev/lib/cmake/shogun/ShogunTargets.cmake:81 (message):
  The imported target "shogun::shogun" references the file

     "/nix/store/a6y70prh981r98hs1plk24r4rqzx335s-shogun-6.1.4-dev/lib/libshogun.so.18.0"

  but this file does not exist.  Possible reasons include:

  * The file was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and contained

     "/nix/store/a6y70prh981r98hs1plk24r4rqzx335s-shogun-6.1.4-dev/lib/cmake/shogun/ShogunTargets.cmake"

  but not all the files it references.

Call Stack (most recent call first):
  /nix/store/a6y70prh981r98hs1plk24r4rqzx335s-shogun-6.1.4-dev/lib/cmake/shogun/ShogunConfig.cmake:29 (include)
  CMakeLists.txt:54 (find_package)
```

This indicates that shogun CMake files are not in the right place and/or are pointing towards the wrong paths.

### Notify maintainers
@edwtjo @smancill

I fixed the issue with shogun. When installing the configuration files, CMake calculates the prefix to the installed library based on the `$dev` output directory, instead of the default `$out`, and I missed it when I split the derivation into multiple outputs. I wonder why it's not a more widespread issue, only a few derivations patch it.

But that `feat` derivation still doesn't build:

```
feat> /tmp/nix-build-feat-0.4.2.drv-0/source/src/pop/op/learn/n_split.cc:216:30: error: no viable overloaded '='
feat>             class_weights(c) = float(
feat>             ~~~~~~~~~~~~~~~~ ^ ~~~~~~
```

This is in macOS, though. Maybe it works on Linux.

--- a/pkgs/applications/science/machine-learning/shogun/default.nix
+++ b/pkgs/applications/science/machine-learning/shogun/default.nix
@@ -6,7 +6,7 @@
   # build
 , cmake
 , ctags
-, python2Packages
+, python3Packages
 , swig
   # math
 , eigen
@@ -30,13 +30,13 @@
 , lp_solve
 , colpack
   # extra support
-, pythonSupport ? true
+, pythonSupport ? false
 , opencvSupport ? false
 , opencv ? null
 , withSvmLight ? false
 }:
 
-assert pythonSupport -> python2Packages != null;
+assert pythonSupport -> python3Packages != null;
 assert opencvSupport -> opencv != null;
 
 assert (!blas.isILP64) && (!lapack.isILP64);
@@ -101,7 +101,7 @@ stdenv.mkDerivation rec {
   ] ++ lib.optional (!withSvmLight) ./svmlight-scrubber.patch;
 
   nativeBuildInputs = [ cmake swig ctags ]
-    ++ (with python2Packages; [ python jinja2 ply ]);
+    ++ (with python3Packages; [ python jinja2 ply ]);
 
   buildInputs = [
     eigen
@@ -121,7 +121,7 @@ stdenv.mkDerivation rec {
     nlopt
     lp_solve
     colpack
-  ] ++ lib.optionals pythonSupport (with python2Packages; [ python numpy ])
+  ] ++ lib.optionals pythonSupport (with python3Packages; [ python numpy ])
     ++ lib.optional opencvSupport opencv;
 
   cmakeFlags = let
@@ -139,7 +139,7 @@ stdenv.mkDerivation rec {
     "-DENABLE_TESTING=${enableIf doCheck}"
     "-DDISABLE_META_INTEGRATION_TESTS=ON"
     "-DTRAVIS_DISABLE_META_CPP=ON"
-    "-DPythonModular=${enableIf pythonSupport}"
+    "-DINTERFACE_PYTHON=${enableIf pythonSupport}"
     "-DOpenCV=${enableIf opencvSupport}"
     "-DUSE_SVMLIGHT=${enableIf withSvmLight}"
   ];

--- a/pkgs/applications/science/machine-learning/shogun/default.nix
+++ b/pkgs/applications/science/machine-learning/shogun/default.nix
@@ -177,6 +177,12 @@ stdenv.mkDerivation rec {
     rm -r $out/share
   '';
 
+  postFixup = ''
+    # CMake incorrectly calculates library path from dev prefix
+    substituteInPlace $dev/lib/cmake/shogun/ShogunTargets-release.cmake \
+      --replace "\''${_IMPORT_PREFIX}/lib/" "$out/lib/"
+  '';
+
   meta = with lib; {
     description = "A toolbox which offers a wide range of efficient and unified machine learning methods";
     homepage = "http://shogun-toolbox.org/";

